qiskit/compiler/assemble.py

```diff
@@ -220,10 +220,12 @@ def _parse_common_args(backend, qobj_id, qobj_header, shots,
             shots = min(1024, max_shots)
         else:
             shots = 1024
-    elif max_shots and max_shots < shots:
+    elif isinstance(shots, int) and max_shots and max_shots < shots:
         raise QiskitError(
             'Number of shots specified: %s exceeds max_shots property of the '
             'backend: %s.' % (shots, max_shots))
+    elif not isinstance(shots, int):
+        raise TypeError('The attribute \'shots\' must be of type int')
```

Comment:
This should raise a QiskitError. A note should be added in the Raises section that this may raise if shots is not right.